internal/testrunner/runners/pipeline/coverage.go

```diff
@@ -115,6 +115,17 @@ func pipelineDataForCoverage(pipeline ingest.Pipeline, stats ingest.PipelineStat
 		return "", "", nil, ingest.PipelineStats{}, fmt.Errorf("pipeline '%s' not installed in Elasticsearch", pipeline.Name)
 	}
 
+	// remove reroute processors if any so the pipeline has the same processors as in the file
+	// reroute processors are added if there are any routing_rules file defined
+	var processors []ingest.ProcessorStats
+	for _, proc := range pstats.Processors {
+		if proc.Type == "reroute" {
+			continue
+		}
+		processors = append(processors, proc)
+	}
+	pstats.Processors = processors
+
```

Comment:
As the stats returned by Elasticsearch contain the `reroute` processors added by elastic-package in a previous step, here it is needed to remove those processors so it contains the same number of processors as in the file.